### ISSUE TITLE
Improve generated files detection

### ISSUE THREAD

**Issue by rayjlinden** [golangci/golangci-lint#1083]
golangci-lint is not ignoring auto generated files like it is supposed to.  The autogenerated files are made by open-api generator.  Here is a part of the top of one of the generated files:

```
/*
 * webhooksio client
 *
 * Hack because Ray is too lazy to write a client by hand.
 *
 * API version: 1.0.0
 */

// Code generated by OpenAPI Generator (https://openapi-generator.tech); DO NOT EDIT.
```

Thank you for creating the issue!

- [x] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [x] Yes, I've searched similar issues on GitHub and didn't find any.
- [x] Yes, I've included all information below (version, config, etc).

Please include the following information:

<details><summary>Version of golangci-lint</summary>

```console
$ golangci-lint --version
# golangci-lint has version 1.26.0 built from 6bd10d0 on 2020-05-01T15:33:57Z
```

</details>

<details><summary>Config file</summary>

```console
$ golangci-lint run --fix --no-config --disable-all -E gofmt -E govet -E errcheck -E staticcheck -E unused -E gosimple -E structcheck -E varcheck -E ineffassign -E deadcode -E typecheck -E gosec ./...
```

</details>

<details><summary>Go environment</summary>

```console
$ go version && go env
go version go1.13 linux/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/rayj/.cache/go-build"
GOENV="/home/rayj/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GONOPROXY="github.com/lindenlab/*"
GONOSUMDB="github.com/lindenlab/*"
GOOS="linux"
GOPATH="/var/tmp/rayj/gocode"
GOPRIVATE="github.com/lindenlab/*"
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/home/rayj/.gvm/gos/go1.13"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/home/rayj/.gvm/gos/go1.13/pkg/tool/linux_amd64"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/var/tmp/rayj/gocode/src/github.com/lindenlab/ecom-webhooks/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build126193616=/tmp/go-build -gno-record-gcc-switches"```

</details>

<details><summary>Verbose output of running</summary>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
DEBU [autogen_exclude] doc of len 0 doesn't contain any of markers: [code generated do not edit autogenerated file]
DEBU [autogen_exclude] file "webhooksio/api_default.go" is generated: false
DEBU [autogen_exclude] doc of len 0 doesn't contain any of markers: [code generated do not edit autogenerated file]
DEBU [autogen_exclude] file "webhooksio/client.go" is generated: false
DEBU [autogen_exclude] doc of len 0 doesn't contain any of markers: [code generated do not edit autogenerated file]
DEBU [autogen_exclude] file "lambda/main.go" is generated: false
webhooksio/client.go:241:18: Error return value of `w.WriteField` is not checked (errcheck)
					w.WriteField(k, iv)
					            ^
webhooksio/client.go:75:6: `atoi` is unused (deadcode)
func atoi(in string) (int, error) {
     ^
webhooksio/client.go:114:6: `typeCheckParameter` is unused (deadcode)
func typeCheckParameter(obj interface{}, expected string, name string) error {
     ^
webhooksio/client.go:152:6: `parameterToJson` is unused (deadcode)
func parameterToJson(obj interface{}) (string, error) {
     ^
webhooksio/client.go:402:6: `reportError` is unused (deadcode)
func reportError(format string, a ...interface{}) error {
     ^
webhooksio/client.go:508:6: `strlen` is unused (deadcode)
func strlen(s string) int {
     ^
webhooksio/client.go:106:6: SA6005: should use strings.EqualFold(a, b) instead of strings.ToLower(a) == strings.ToLower(b) (staticcheck)
		if strings.ToLower(a) == strings.ToLower(needle) {
		   ^
```

</details>

**Comment by rayjlinden** [comment]
Looking through how golangci-lint looks for auto-generated code - it appears the C style comment block above the code gen line is the issue.  This causse the loop to break and it never sees the auto-gen line.  In fact, if I removed that first comment from the generated file - it golangci-lint correctly determined it was an auto-generated file.

It is this code here that seems to be the problem:
https://github.com/golangci/golangci-lint/blob/eeff3902d440513f5598391d6589c779d39af796/pkg/result/processors/autogenerated_exclude.go#L133-L142

**Comment by ernado** [comment]
As mentioned in https://github.com/golang/go/issues/13560#issuecomment-288457920:
> The text may appear anywhere in the file.

So your file is correct and we should improve detection logic.
Thank you for your report, I'll just slightly adjust issue title.

**Comment by ldez** [comment]
Fixed by  #1161
